hw/ip/rv_core_ibex/doc/boot-rom-patching.md

```diff
@@ -175,51 +175,67 @@ In this case, the dedicated SRAM can be locked after being loaded so that the pa
+This re-routing is handled by a logic block that sits between the Ibex core and TL-UL crossbar.
+The block is configured by 32 sets of 4 match registers:
+- [`IBUS_ADDR_MATCHING`](../data/rv_core_ibex.hjson#ibus_addr_matching_0)
+- [`IBUS_ADDR_EN`](../data/rv_core_ibex.hjson#ibus_addr_en_0)
+- [`IBUS_REMAP_ADDR`](../data/rv_core_ibex.hjson#ibus_remap_addr_0).
+- [`IBUS_REGWEN`](../data/rv_core_ibex.hjson#ibus_regwen_0).
 
-A patch match register is structured as follows.
-The patch applies to a small region of code (4, 8, 16 or 32 bytes), starting at a given address.
-This region is encoded in a field called `M_BASE`.
-The destination of the redirection is given by its address, encoded in a field called `R_BASE`.
+The patch loader is responsible for configuring those registers for each patch in the patch table.
+Each entry in the OTP ROM patch table is composed of two 32 bit words with the following layout:
 
-- If the source region has address `yyy...yyy`, the `M_BASE` field is:
+![Patch match register layout](patch-match.svg)
+
+- The first 32 bits describe the patch match configuration:
+  - `M_BASE` is the base address of the original code in ROM to be patched.
+  - `P_SIZE` is the size, in `DWORD`s, of the code region to be patched.
+    ROM patches are small code regions that can be 4, 8 16 or 32 bytes long, and thus the only valid values for `P_SIZE` are 1, 2, 4 and 8.
+    Setting `P_SIZE` to any other value will make the patch invalid and the patch loader will not enable it.
+  - `L` is the the lock bit.
+    When this is set, the system will disallow any writes to the match registers.
+- The second 32-bit word contains the remapping address base:
+  - `R_BASE` is the remapped address for the patched code region.
+    This is the SRAM address where the patch loader loads the code patch.
+    Once the patch loader enables a patch region, instruction fetches that match an address in the `[M_BASE..MBASE + P_SIZE]` interval will be replaced by the instruction at the same offset within the `[R_BASE..RBASE + P_SIZE]` instead.
+
+When applying ROM patches, the patch loader configures the Ibex instruction re-routing logic for each patch region, as follows:
+
+1. `IBUS_ADDR_MATCHING` is set to `(M_BASE & ~(P_SIZE - 1)) | ((P_SIZE - 1) >> 1)` in order for the Ibex remapping logic to form the ROM patch matching mask for that region.
+   See next section for more details.
+1. `IBUS_REMAP_ADDR` is set to `R_BASE`.
+1. `IBUS_ADDR_EN` is set to 1 if the patch loader evaluates the patch to be a valid one.
+1. `IBUS_REGWEN` is set to 0 if `L` is set to 1.
+
+#### Patch Match Logic Details
+
+As described in the previous section, the `IBUS_ADDR_MATCHING` register is configured by the patch loader with a value that aggregates both the ROM patch region base address and the region size.
+
+- If the OTP patch table first word `M_BASE` value is `yyy...yyy`, then the `IBUS_ADDR_MATCHING` register is configured as follows:
```

Comment:
Maybe this would be easier to read as "If the `M_BASE` value for the first entry in the OTP patch table ..." ?